Fix Weblinks target issue with template override

Amitesh007z opened the pull request:
Pull Request for Issue #399

Summary of Changes
Implemented a template override for com_finder to ensure Weblinks open in a new tab when the global configuration (Web Links > Target > Open in new window) is set.
Modified default_result.php under the Finder component to adjust the target="_blank" behavior for Weblinks.
Ensured that links generated in the search results follow the expected structure

Testing Instructions
Go to Global Configuration > Web Links > Target and set it to Open in new window.
Use Smart Search (Finder) to search for a Weblink that has Target set to Global.
Click on the link in the search results.
Observe whether the Weblink opens in a new tab (expected behavior).

Expected result
Weblink should open in a new tab

https://github.com/user-attachments/assets/6e27935a-175e-40d8-97ce-c29a11488f48



Actual result
Opens in parent window.

alikon commented on src/templates/cassiopeia/html/mod_custom/banner.php:
why you need this and the others ??

Amitesh007z replied:
yeah sir the only main file which has changes for template override is  html/com_finder/search/default_result.php , i added these unrelated files as , there was no templates folder in repo , thought it would be odd if i add this alone sir , sso i just added all together , i guess it would be little confusing with large number of unrelated files sir , very sorry for that , can i just **keep html/com_finder/search/default_result.php** this file  and **remove all other files** as it only has main template override changes sir ?

alikon reviewed (changes requested):
too much unrelated files

Amitesh007z commented:
> too much unrelated files

Hi sir , actually the main and only file to be added is html/com_finder/search/default_result.php as it had main change , but as  there was no templates folder in this forked repo(as it wasnt locally present) ,  i went with adding whole template folder (from my locally setup repo ) sir , i guess it would be little confusing with large number of unrelated files sir , very sorry for that , can i just add html/com_finder/search/default_result.php this file as it only  has main template override changes sir ?

alikon commented:
please check code style https://ci.joomla.org/joomla-extensions/weblinks/2062/1/5

also move the pr to branch 5.x-dev

joomdonation commented:
I haven't looked at the issue details but this is wrong approach. You cannot solve an issue by adding code does not belong to the extension. If it is needed, the PR must be made in https://github.com/joomla/joomla-cms, but I'm unsure if this is the right solution

Amitesh007z commented:
> please check code style https://ci.joomla.org/joomla-extensions/weblinks/2062/1/5
> 
> also move the pr to branch 5.x-dev

yes sir , i have moved the PR to branch 5.x-dev and corrected styling , 
Soory for the convenience caused sir

joomdonation commented:
As stated in my earlier comment https://github.com/joomla-extensions/weblinks/pull/605#issuecomment-2783455666, this approach is wrong. We could not implement a layout override for a different component in weblinks to support this feature

Unless the Smart Search component gives us a way to allow open link in popup, this is not doable.

I'm closing this PR for now. Thanks for your effort.

Amitesh007z commented:
hi @joomdonation  , I've re-evaluated the approach based on the feedback  and agree that overriding layout behavior of another component (Smart Search) isn't good 

Instead, I propose  which i discussed in comments of the issue (https://github.com/joomla-extensions/weblinks/issues/399#issuecomment-2764505233) i.e adding a new admin option in the Web Links component settings:
"Open search result in new window" (Yes/No)
When Yes, the Web Link will open in a new tab directly from search results.
When No (default), the current behavior remains — the user is taken to the detail page.

This approach:
Keeps current behavior unchanged unless configured otherwise.Stays within the boundaries of the Web Links component (no layout overrides).Improves UX for certain use cases (like link directories).

I’d love to hear your feedback and am happy to update the PR accordingly.

joomdonation commented:
> I’d love to hear your feedback and am happy to update the PR accordingly.

How would you handle that because the behavior (Open Search Result) is controlled by Smart Search component (from Joomla core), not from Weblinks?

Amitesh007z commented:
> > I’d love to hear your feedback and am happy to update the PR accordingly.
> 
> How would you handle that because the behavior (Open Search Result) is controlled by Smart Search component (from Joomla core), not from Weblinks?

Hi @joomdonation , Very sorry for last response , 

To implement the feature without modifying the core Smart Search component, I propose creating a layout override specifically for the Smart Search results. This would allow us to handle Web Links results conditionally, ensuring they open in a new tab based on the setting in the Web Links component.Like In this override, you can check if the result is from the Web Links component and adjust the link behavior accordingly, 

will be happy to hear your feedback sir

joomdonation commented:
@Amitesh007z We won't do that. We will leave it as how it is for now until Smart Search component allows us to do that. We really depends on Smart Search in this case, no reasons for we to support this feature if all other Joomla components and Smart Search itself does not support it.

If someone needs this feature, they should implement the override themself. You can ignore this issue
